src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -260,6 +306,10 @@ def serialize(
                 for child in spans:
                     if child["span_id"] in error_map:
                         previous_event["errors"].extend(error_map.pop(child["span_id"]))
+                    # We need to connect back to an existing orphan trace
+                    if has_orphans and child["span_id"] in results_map:
+                        previous_event["children"].extend(results_map.pop(child["span_id"]))
+                        self.update_generations(previous_event)
```

Comment:
This seems like an exponential update in the worst case. For example, we first see `event A`, then we see `event B` and make that a parent and update the generations. Then we see `event C` and make that a parent of `event C`.

Can the update just be run once at the very end?